### ISSUE TITLE
DEVC layer invalid

### ISSUE THREAD

**Issue by Gekke-Jakke** [larshp/abapGit#1020]
When importing a package an exception was raised: Error from IF_PACKAGE->SET_ALL_ATTRIBUTES( 8 ).

Issue can be found in:
![image](https://user-images.githubusercontent.com/2744548/31929122-f40160fe-b89a-11e7-9709-20178bc7f0c7.png)

Had to solve this issue with two corrections using the debugger:
![image](https://user-images.githubusercontent.com/2744548/31929135-0bd27d08-b89b-11e7-9d57-9e31a05abe35.png)

Missing namespace is related to de serialization method which clears the namespace:
![image](https://user-images.githubusercontent.com/2744548/31929167-2d5d784c-b89b-11e7-90d7-09be128acc78.png)

The missing superpackage issue I couldn't find right away, but probably is also related to the serialization of the package. (double checked the XML file, super package is missing)

**Comment by flaiker** [comment]
Sorry for the delay, I'll provide a fix this weekend.
Some notes:
- Parent package not being serialized in the xml is intended, as it may be different in each system because of the PREFIX folder logic ([docs](http://docs.abapgit.org/ref-dot-abapgit.html)). It should however of course be determined before calling the package API.
- Namespaces _should_ probably be serialized to the xml and not cleared, as you noted. They are very difficult to test for me, because I don't have access to a system that has them set up and where I could install abapGit on (afaik there's no way to do it on AS ABAP developer edition without your company providing the namespace key).
I think I might have added the namespace-clear-statement at a later point because it caused some problems with `/0CUST/`-namespace and Z-objects. I'll have a look.

**Comment by flaiker** [comment]
`SUPERPACKAGE_IN_TDEVC` seems to only be used internally (at least on my release) and not needed in `SET_ALL_ATTRIBUTES` as the methods that use it will set it from `PARENTCL`, which `lcl_sap_package->create_child` _should_ set correctly.

Can you check if just adding the namespace to the serialization fixes your problem, or if more is needed?

flaiker@b7b0d27
```diff
diff --git a/src/zabapgit_object_devc.prog.abap b/src/zabapgit_object_devc.prog.abap
index 25b55615..0cd5c604 100644
--- a/src/zabapgit_object_devc.prog.abap
+++ b/src/zabapgit_object_devc.prog.abap
@@ -338,8 +338,7 @@ CLASS lcl_object_devc IMPLEMENTATION.
     ENDIF.
 
     " Clear things related to local installation package
-    CLEAR: ls_package_data-namespace,
-           ls_package_data-dlvunit,
+    CLEAR: ls_package_data-dlvunit,
            ls_package_data-pdevclass.
 
     " Not usable on customer systems
```

**Comment by Gekke-Jakke** [comment]
@flaiker same issue still occurs, but I noticed that when I jump back to the call set_all_attributes (using the debugger) after sy-subrc = 8 and redo the same logic (without changing any parameters) the call returns with a sy-subrc = 0 and everything seems to be fine at first.

But when I check the package details the transport layer is removed:
![image](https://user-images.githubusercontent.com/2744548/32134807-1f866ba4-bbf5-11e7-9d51-1c0d91bc5c1e.png)

I'll try to dig into this issue deeper later on and keep you posted. But already wanted to give you a heads-up!

**Comment by aferngas** [comment]
This works for me:

Context block: LCL_OBJECT_DEVC
```
METHOD lif_object~deserialize.
    DATA: li_package         TYPE REF TO if_package,
          ls_package_data    TYPE scompkdtln,
          ls_data_sign       TYPE scompksign,
          lv_changeable      TYPE abap_bool,
          lt_usage_data      TYPE scomppdata,
          lt_permissions     TYPE tpak_permission_to_use_list,
          li_usage           TYPE REF TO if_package_permission_to_use,
          ls_usage_data_sign TYPE scomppsign,
          ls_save_sign       TYPE paksavsign.

    FIELD-SYMBOLS: <ls_usage_data> TYPE scomppdtln.

    mv_local_devclass = iv_package.

    io_xml->read(
      EXPORTING
        iv_name = 'DEVC'
      CHANGING
        cg_data = ls_package_data ).

    li_package = get_package( ).

    " Swap out repository package name with the local installation package name
    ls_package_data-devclass = mv_local_devclass.
```
New line:
```
    ls_package_data-pdevclass = li_package->transport_layer.
```

**Comment by larshp** [comment]
thanks, feel free to submit a pull request 😄

**Comment by fidley** [comment]
I faced this issue yesterday, and in my case the problem was that when I create a package in my company then we have Z-transport layer setup. ABAPGIT is passing empty transport layer to set_all_atrributes method but in the same time , the transport layer is set to be updated.

CL_PACKAGE then throws an error.
![2018-02-28_09h57_37](https://user-images.githubusercontent.com/7912195/36779608-dbcdaba2-1c70-11e8-8da0-d97799b14f09.png)


I think commenting the     

> ls_data_sign-pdevclass        = abap_true. 


![2018-02-28_09h58_54](https://user-images.githubusercontent.com/7912195/36779620-e6997a48-1c70-11e8-8146-ad867862a729.png)

should solve the problem, but I don't know what side effects it would bring in other places. 
Anyway it worked for me when under debugger I've updated ew_wbo_data-pdevclass to be equal to  data_buffer-pdevclass during runtime of abapgit.
